src/main/kotlin/com/avail/descriptor/pojos/PojoFinalFieldDescriptor.kt

```diff
@@ -0,0 +1,281 @@
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice, this
+ *     list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *
+ *  * Neither the name of the copyright holder nor the names of the contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.avail.descriptor.pojos
+
+import com.avail.annotations.AvailMethod
+import com.avail.descriptor.Descriptor
+import com.avail.descriptor.pojos.PojoFinalFieldDescriptor.ObjectSlots.*
+import com.avail.descriptor.representation.A_BasicObject
+import com.avail.descriptor.representation.AvailObject
+import com.avail.descriptor.representation.AvailObject.Companion.multiplier
+import com.avail.descriptor.representation.Mutability
+import com.avail.descriptor.representation.ObjectSlotsEnum
+import com.avail.descriptor.types.*
+import com.avail.descriptor.types.BottomTypeDescriptor.*
+import com.avail.descriptor.types.VariableTypeDescriptor.variableReadWriteType
+import com.avail.descriptor.variables.VariableDescriptor
+import com.avail.exceptions.AvailErrorCode
+import com.avail.exceptions.AvailErrorCode.E_CANNOT_MODIFY_FINAL_JAVA_FIELD
+import com.avail.exceptions.AvailErrorCode.E_JAVA_MARSHALING_FAILED
+import com.avail.exceptions.AvailRuntimeException
+import com.avail.exceptions.VariableSetException
+import com.avail.serialization.SerializerOperation
+import com.avail.utility.json.JSONWriter
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier
+import java.util.*
+
+/**
+ * A `PojoFinalFieldDescriptor` is an Avail [variable][VariableDescriptor] that
+ * facilitates access to the instance Java [Field] of a particular
+ * [pojo][PojoDescriptor] or the static field of a particular
+ * [pojo&#32;type][PojoTypeDescriptor]. It supports the same protocol as any
+ * other variable, but reads and writes are of the pojo's field.
+ *
+ * It leverages the fact that the field is [final][Modifier.isFinal] by caching
+ * the value and not retaining the reflected field directly.
+ *
+ * @constructor
+ *
```

Comment:
missing constructor comment

` * Construct a new {@link PojoFinalFieldDescriptor}.`

Reply:
Intentionally omitted.  The class comment says what it is, and the constructor just says it creates one.